app/graphql/mutations/profile/associate_rules.rb

```diff
@@ -33,6 +34,16 @@ def audit_mutation(profile, added, removed)
           " #{added} rules added, #{removed} rules removed"
         )
       end
+
+      def prepare_params(args)
+        if args[:rule_ids]
+          { ids: args[:rule_ids] }
+        elsif args[:rule_ref_ids]
+          { ref_ids: args[:rule_ref_ids] }
+        else
+          raise ArgumentError, 'Missing argument identifying rules'
```

Comment:
Yeah, I was not sure if this kind of error handling is okay but I assume it is. Gonna test it

Reply:
I think it's ok, but maybe the error type isn't quite right. Maybe `ActionController::ParameterMissing` would be more appropriate, since it's a required parameter (either rule_ids or rule_ref_ids). This is also handled in our [application_controller](https://github.com/RedHatInsights/compliance-backend/blob/master/app/controllers/application_controller.rb#L49).